### PR TITLE
math/extended_euclidean_algorithm: add doctests and support for negative numbers

### PR DESCRIPTION
**Pull request by pikulet**
### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by dhruvmanila** [comment]
As mentioned in #2630, can you make the appropriate changes for the algorithm to handle negative numbers?

**Comment by pikulet** [comment]
done

File: maths/extended_euclidean_algorithm.py [hunk 1]
```diff
@@ -3,69 +3,97 @@
...
-# @Last modified by:   PatOnTheBack
-# @Last modified time: 2019-07-05
+# @Last modified by:   pikulet
+# @Last modified time: 2020-10-02
 
 import sys
 
 
-def extended_euclidean_algorithm(m, n):
+def sign(n: int) -> int:
```
Comment (dhruvmanila) [inline maths/extended_euclidean_algorithm.py:19]: Is this function really necessary?

File: maths/extended_euclidean_algorithm.py [hunk 2]
```diff
@@ -3,69 +3,97 @@
...
-        t = b_prime
-        b_prime = b
-        b = t - q * b
-
-    pair = None
-    if m > n:
-        pair = (a, b)
+    old_r, r = a, b
+    old_s, s = 1, 0
+    old_t, t = 0, 1
```
Comment (dhruvmanila) [inline maths/extended_euclidean_algorithm.py:72]: Please use descriptive variable names everywhere.

Comment (pikulet) [reply maths/extended_euclidean_algorithm.py]: I have changed to 

r --> remainder
s--> coeff_a
t --> coeff_b

File: maths/extended_euclidean_algorithm.py (hunk 1 above)
Comment (pikulet) [reply maths/extended_euclidean_algorithm.py]: I have made changes that does not require sign to do negative number correction

File: maths/extended_euclidean_algorithm.py [hunk 3]
```diff
@@ -3,69 +3,81 @@
...
-# @Last modified by:   PatOnTheBack
-# @Last modified time: 2019-07-05
+# @Last modified by:   pikulet
+# @Last modified time: 2020-10-02
 
 import sys
 
 
-def extended_euclidean_algorithm(m, n):
+def extended_euclidean_algorithm(a: int, b: int) -> (int, int):
```
Comment (dhruvmanila) [inline maths/extended_euclidean_algorithm.py:19]:
```suggestion
def extended_euclidean_algorithm(a: int, b: int) -> Tuple[int, int]:
```
Please put this with other imports: `from typing import Tuple`

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

**Review by dhruvmanila — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**